api/lib/db-interface.js

```diff
@@ -226,6 +243,10 @@ DI.saveTransaction = function(transaction, callback) {
                 + err);
 
     (callback || noop)(err);
+  })
+  .finally(function() {
+    self.lock[txData.client_resource_id] = undefined;
```

Comment:
Memory leaks can crash a process, you should be more concerned about that than optimizations.